### PR TITLE
Get master nodes but useless code

### PR DESCRIPTION
**Pull request by wgr523**
They are useless since they are not rebased.

### CODE REVIEW

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 1]
```diff
@@ -594,8 +583,10 @@ func (x *XDPoS_v2) onVotePoolThresholdReached(chain consensus.ChainReader, poole
 				- Use the above xdc address to check against the master node(For the running epoch)
 		2. Broadcast(Not part of consensus)
 */
-func (x *XDPoS_v2) VerifyTimeoutMessage(timeoutMsg *utils.Timeout) (bool, error) {
-	return x.verifyMsgSignature(utils.TimeoutSigHash(&timeoutMsg.Round), timeoutMsg.Signature)
+func (x *XDPoS_v2) VerifyTimeoutMessage(chain consensus.ChainReader, timeoutMsg *utils.Timeout) (bool, error) {
+	//TODO: question: Do we need lock here?
+	masternodes := x.GetMasternodesAtRound(chain, timeoutMsg.Round, chain.CurrentHeader())
+	return x.verifyMsgSignature(masternodes, utils.TimeoutSigHash(&timeoutMsg.Round), timeoutMsg.Signature)
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go:589]: Mark

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 2]
```diff
@@ -530,7 +518,8 @@ func (x *XDPoS_v2) VerifyVoteMessage(vote *utils.Vote) (bool, error) {
 			2. Verify blockInfo
 			3. Broadcast(Not part of consensus)
 	*/
-	return x.verifyMsgSignature(utils.VoteSigHash(vote.ProposedBlockInfo), vote.Signature)
+	masternodes := x.GetMasternodesByHash(chain, vote.ProposedBlockInfo.Hash)
+	return x.verifyMsgSignature(masternodes, utils.VoteSigHash(vote.ProposedBlockInfo), vote.Signature)
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go:522]: Mark

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 3]
```diff
@@ -912,15 +903,14 @@ func (x *XDPoS_v2) signSignature(signingHash common.Hash) (utils.Signature, erro
...
-func (x *XDPoS_v2) verifyMsgSignature(signedHashToBeVerified common.Hash, signature utils.Signature) (bool, error) {
+func (x *XDPoS_v2) verifyMsgSignature(masternodes []common.Address, signedHashToBeVerified common.Hash, signature utils.Signature) (bool, error) {
 	// Recover the public key and the Ethereum address
 	pubkey, err := crypto.Ecrecover(signedHashToBeVerified.Bytes(), signature)
 	if err != nil {
 		return false, fmt.Errorf("Error while verifying message: %v", err)
 	}
 	var signerAddress common.Address
 	copy(signerAddress[:], crypto.Keccak256(pubkey[1:])[12:])
-	masternodes := x.getCurrentRoundMasterNodes()
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go:923]: Mark

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 4]
```diff
@@ -1053,3 +1043,185 @@ func (x *XDPoS_v2) GetProperties() (utils.Round, *utils.QuorumCert, *utils.Quoru
...
+		if epochSwitchInfo.EpochSwitchParentBlockInfo.Round/utils.Round(x.config.Epoch) == round/utils.Round(x.config.Epoch) {
+			return x.GetMasternodesByHash(chain, epochSwitchInfo.EpochSwitchParentBlockInfo.Hash)
+		} else {
+			log.Error("[getMasternodesAtRound] should not get masternodes at two epoch ago!")
+			return []common.Address{}
+		}
+	}
+	// if this round enters a brand new epoch (no epoch switch block to use)
+	return x.CalcMasternodes(chain, round, currentHeader)
+}
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go]: Mark

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 5]
```diff
@@ -1053,3 +1043,185 @@ func (x *XDPoS_v2) GetProperties() (utils.Round, *utils.QuorumCert, *utils.Quoru
...
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go:1211]: Mark

**Comment by liam-lai** [comment]
doesn't need

**Pull request closed without merging**